dev/config/src/dependencies.ts

```diff
@@ -23,6 +25,88 @@ async function getPackageDir(packageName: string): Promise<string> {
     return packageDir.trim()
 }
 
+/**
+ * Get the tsconfig paths for a package
+ * @param tsConfigPath
+ * @param ignorePatterns
+ * @param tsConfigPaths
+ */
+export function getTsConfigs(
+    tsConfigPath: string,
+    ignorePatterns: RegExp[] = [],
+    tsConfigPaths: string[] = [],
+    initial = false
+): string[] {
+    let tsConfigs = [...tsConfigPaths]
+    const references = JSON.parse(fs.readFileSync(tsConfigPath).toString()).references
+    if (!tsConfigs.includes(tsConfigPath)) {
+        const ignore =
+            ignorePatterns && ignorePatterns.length > 0 ? new RegExp(`${ignorePatterns.join('|')}`) : undefined
+        if (!initial) {
+            tsConfigs.push(tsConfigPath)
+        }
+        if (references) {
+            for (const reference of references) {
+                // ignore the packages we don't want to bundle
+                if (ignore) {
+                    if (ignore.test(reference.path)) {
+                        continue
+                    }
+                }
+                // remove tsconfig.*.json from the path and get the path to the new directory via the reference path
+                let refTSConfigPath = path.resolve(tsConfigPath.replace(tsConfigRegex, ''), reference.path)
+                if (!refTSConfigPath.endsWith('.json')) {
+                    refTSConfigPath = path.resolve(refTSConfigPath, 'tsconfig.json')
+                }
+                const newTsConfigs = getTsConfigs(refTSConfigPath, ignorePatterns, tsConfigs)
+                if (newTsConfigs.length > 0) {
+                    tsConfigs = [...new Set(tsConfigs.concat(newTsConfigs))]
+                }
```

Comment:
Readability here could be a bit better - maybe worth 10 min refactor if it's not temp stuff?

Reply:
I've used filter and map but kept the recursive reference within the function that's being called recursively.

Reply:
I've removed the map as the result is not used, on the recommendation of @withmaverick